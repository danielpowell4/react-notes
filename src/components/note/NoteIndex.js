import React from 'react';
import {NoteItem} from './NoteItem'

export const NoteIndex = (props) => {
  const noteItems = (props) => {

    return ( props.notes.map(note => <NoteItem key={note.id}
                                               {...note}
                                               handleToggleEdit={props.handleToggleEdit}
                                               handleRemove={props.handleRemove}
                                               handleUpdateContent={props.handleUpdateContent}
                                               handleEditSubmit={props.handleEditSubmit}/> ) )
  }

  return (
    <div className="Note-Index">
      <ul>
        {noteItems(props)}
      </ul>
    </div>
  )
}

  NoteIndex.propTypes = {
    handleEditSubmit: React.PropTypes.func.isRequired,
    handleRemove: React.PropTypes.func.isRequired,
    handleToggleEdit: React.PropTypes.func.isRequired,
    handleUpdateContent: React.PropTypes.func.isRequired,
    notes: React.PropTypes.array.isRequired
  }
