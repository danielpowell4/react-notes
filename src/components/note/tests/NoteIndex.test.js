import React from 'react';
import renderer from 'react-test-renderer';
import {NoteIndex} from '../NoteIndex';

/* NoteIndex snapshot tests */

const noteIndexSampleProps = {
  handleEditSubmit: function(){},
  handleRemove: function(){},
  handleToggleEdit: function(){},
  handleUpdateContent: function(){},
  notes: [{id: 1, isEditting: false, content: 'static note toBe'}, {id:2, isEditting: true, content: 'editting note toBe'}]
}

test('NoteIndex renders both style of notes correctly', () => {
  const tree = renderer.create(
    <NoteIndex {...noteIndexSampleProps} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
