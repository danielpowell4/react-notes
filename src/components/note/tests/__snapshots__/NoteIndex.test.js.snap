exports[`test NoteIndex renders both style of notes correctly 1`] = `
<div
  className="Note-Index">
  <ul>
    <li>
      <span
        className="delete-item">
        <a
          href="#"
          onClick={[Function]}>
          X
        </a>
      </span>
      <span
        onClick={[Function]}>
        static note toBe
      </span>
    </li>
    <form
      onSubmit={[Function]}>
      <input
        onChange={[Function]}
        type="text"
        value="editting note toBe" />
    </form>
  </ul>
</div>
`;
